Lesson15/Busra/busra.js

```diff
@@ -0,0 +1,508 @@
+    console.log("This text is spam.");
+  }else{
+    console.log("This text is not spam.");
+  }
+}
+isSpam("Bu TEXT bir SPAM içerir");
+isSpam("bu TEXT herhangi bir sp*m içermez");
+
+
+
+/*
+23. Extract Initials
+   - Define a function `getInitials(fullName)` that uses .split() to get each name part,
+     then logs the capitalized first letter of each.
+   - Example: "John Doe" -> "J.D."
+*/
+function getInitials(fullName){
+  const nameParts = fullName.split(' ');
+  const initials = [];
+
+  for(let i = 0; i < nameParts.length; i++){
+    const initial = nameParts[i].charAt(0).toUpperCase();
+    initials.push(initial);
+  }
+   const result = initials.join('.')+'.';
+   console.log(result);
+}
+getInitials("busra demirkesen");
+
+
+
+
+/*
+24. Switch: Month to Season
+   - Define a function `getSeason(monthNum)` (1-12). Use switch or if-else:
+     - 12, 1, 2  -> "Winter"
+     - 3, 4, 5   -> "Spring"
+     - 6, 7, 8   -> "Summer"
+     - 9, 10, 11 -> "Autumn"
+   - Log the season or "Invalid month" if out of range.
+*/
+
+function getSeason(monthNum) {
+  switch (monthNum) {
+    case 12:
+    case 1:
+    case 2:
+      console.log("winter");
+      break;
+    case 3:
+    case 4:
+    case 5:
+      console.log("spring");
+      break;
+    case 6:
+    case 7:
+    case 8:
+      console.log("summer");
+      break;
+    case 9:
+    case 10:
+    case 11:
+      console.log("autumn");
+      break;
+    default:
+      console.log("Invalid month");
+  }
+}
+
+getSeason(7);
+
+
+/*
+25. Check If String Contains Number
+   - Define a function `containsNumber(str)` that uses a loop or a method like
+     .match() to check if there's any digit in the string.
+   - Log "Contains number" or "No number found".
+*/
+
+function containsNumber(str){
+  if(str.match(/\d/g)){
+      console.log("Contains number");
+  }else{
+    console.log("No number found");
+  }
+
+}
+containsNumber("busra 34 yaşında");
+containsNumber("busra yaşında");
+
+/*
+26. Pad a String
+   - Define a function `padString(str, maxLength)` that if str.length < maxLength,
+     uses .padEnd() or .padStart() to make the string reach maxLength with '*'.
+   - Log the padded string.
+*/
+
+function padString(str, maxLength){
+  if(str.length<maxLength){
+    str=str.padEnd(maxLength,'*');
+  }
+  console.log(str);
+}
+
+padString("busra",15);
+
+
+/*
+27. If-Else: Voting Eligibility
+   - Define a function `canVote(age)` that logs:
+     - "Can vote" if age >= 18
+     - "Too young to vote" otherwise
+*/
+function canVote(age){
+  if(age>18){
+    console.log("Can vote");
+  }
+  else{
+    console.log("Too young to vote");
+  }
+}
+canVote(17);
+canVote(25);
+
+
+
+/*
+28. Reverse Words in a Sentence
+   - Define a function `reverseWords(sentence)` that:
+     - Splits the sentence by spaces
+     - Reverses each word individually
+     - Joins them back
+   - Log the result.
+*/
+
+function reverseWords(sentence){
+  const reverseSentence = [];
```

Comment:
Use let since you're going to change `reverseSentence` later on.

```suggestion
  let reverseSentence = [];
```